### PR TITLE
Multi php remi grain

### PR DESCRIPTION
**Pull request by didaurel**

### CODE REVIEW

**Comment by gervasek** [comment]
cloud9RestartPhp alias in ~/.bashrc.d/00-devops.lamp.sh must be updated:
```
$ cloud9RestartPhp
php-fpm: unrecognized service
```

**Comment by gervasek** [comment]
Symlinks for php71 config are not correct:
```
$ ll conf.d/php71/*
lrwxrwxrwx 1 ec2-user ec2-user 30 2019-03-20 15:21:24 conf.d/php71/php.d -> /opt/remi/php71/root/etc/php.d
lrwxrwxrwx 1 ec2-user ec2-user 35 2019-03-20 15:21:24 conf.d/php71/php-fpm.d -> /opt/remi/php71/root/etc/php-fpm.d/
```

```
$ ll /opt/remi/php71/root/etc/php.d
/bin/ls: cannot access /opt/remi/php71/root/etc/php.d: No such file or directory
```
Apparently for 7.1 conf directory is under: `/etc/opt/remi/php71/php.d`

**Comment by gervasek** [comment]
it should be good to add symlink to mysql conf folder in conf.d:
```
ln -s /etc/my.cnf.d/ /home/ec2-user/environment/conf.d/my.cnf.d
```
(and ensure ec2-user have permission to edit files in this directory)

**Comment by gervasek** [comment]
Add in file 00-devops.ini
```
;#############################
;#  Do not edit this file.   #
;#  File managed by salt.    #
;#############################
```

File: salt/lamp/php-fpm/files/00-devops.ini [hunk 1]
```diff
@@ -8,7 +8,7 @@ max_input_time = 240
 ; ERRORS
 error_reporting = E_ALL
 display_errors = on
-error_log = /var/log/php-fpm/error.log
+error_log = /var/log/php-fpm/error{{ php_version }}.log
```
Comment (gervasek) [inline salt/lamp/php-fpm/files/00-devops.ini]: Something wrong with php logs, not in `/var/log/php-fpm/` but in:
```
/opt/remi/php56/root/var/log/php-fpm/error.log
/var/opt/remi/php71/log/php-fpm/error.log
```

Comment (gervasek) [reply salt/lamp/php-fpm/files/00-devops.ini]: OK sorry, error logs for "php cli" and "php from httpd" is in `/var/log/php-fpm/` folder.
But the log for php56-php-fpm seems to be in `/opt/remi/php56/root/var/log/php-fpm/error.log`
This logfile is configured in default php-fpm.conf: `/opt/remi/php56/root/etc/php-fpm.conf`
```
; Default Value: /opt/remi/php56/root/var/log/php-fpm.log
error_log = /opt/remi/php56/root/var/log/php-fpm/error.log
```
I'm not sure, but maybe it should be better to redefine in /var/log/php-fpm on our php-fpm.conf

**Comment by gervasek** [comment]
it would be nice to add symlink to mysql log in folder in `/home/ec2-user/environment/log` folder

(and ensure ec2-user have permission to edit files in this directory)

File: salt/lamp/php-fpm/init.sls [hunk 2]
```diff
@@ -59,18 +115,34 @@ php-extensions-sup-71:
...
 
-create_php_log_dir:
-  file.directory:
-    - name: '/home/ec2-user/environment/log'
+##############
+# CONF LINKS #
+##############
+/home/ec2-user/environment/conf.d/php{{ php_version }}/php-fpm.d:
+  file.symlink:
+    - target: /opt/remi/php{{ php_version }}/root/etc/php-fpm.d/
```
[the remark this answers is not in the document]
Comment (didaurel) [reply salt/lamp/php-fpm/init.sls]: The lines has been deleted.

File: salt/lamp/php-fpm/files/php-fpm.conf [hunk 3]
```diff
@@ -1,7 +1,6 @@
...
+#############################
+#  Do not edit this file.   #
+#  File managed by salt.    #
+#############################
 
-<FilesMatch \.php$>
-	SetHandler proxy:fcgi://php-fpm
-</FilesMatch>
\ No newline at end of file
+SetHandler proxy:unix:/var/opt/remi/php{{ php_version }}/run/php-fpm/default.sock|fcgi://localhost
```
[the remark this answers is not in the document]
Comment (didaurel) [reply salt/lamp/php-fpm/files/php-fpm.conf]: Done

File: salt/lamp/php-fpm/files/www.conf [hunk 4]
```diff
@@ -1,3 +1,8 @@
+;#############################
+;#  Do not edit this file.   #
+;#  File managed by salt.    #
+;#############################
+;{set php_version = salt['pillarhp-fpm:version','56') %}
```
[the remark this answers is not in the document]
Comment (didaurel) [reply salt/lamp/php-fpm/files/www.conf]: Done

File: salt/lamp/php-fpm/init.sls [hunk 5]
```diff
@@ -59,18 +115,34 @@ php-extensions-sup-71:
...
     - group: ec2-user
     - user: ec2-user
+    - makedirs : true
 
+/home/ec2-user/environment/conf.d/php{{ php_version }}/php.d:
+  file.symlink:
+    - target: /opt/remi/php{{ php_version }}/root/etc/php.d
```
[the remark this answers is not in the document]
Comment (didaurel) [reply salt/lamp/php-fpm/init.sls]: done

File: salt/lamp/php-fpm/files/00-devops.ini (hunk 1 above)
Comment (didaurel) [reply salt/lamp/php-fpm/files/00-devops.ini]: Done

File: salt/lamp/bashrc/files/00-devops.lamp.sh [hunk 6]
```diff
@@ -1,6 +1,6 @@
 
-alias cloud9RestartApache='sudo service httpd restart'
-alias cloud9RestartMysql='sudo service mysql restart'
-alias cloud9RestartPhp='sudo service php-fpm restart'
+alias cloud9RestartApache="sudo service httpd restart"
+alias cloud9RestartMysql="sudo service mysql restart"
+alias cloud9RestartPhp="sudo service php{{ php_version }}-php-fpm restart"
```
Comment (gervasek) [inline salt/lamp/bashrc/files/00-devops.lamp.sh]: With this solution we can only restart 1 php version...
Proposal:
- 1 alias by php version (cloud9RestartPhp56, cloud9RestartPhp71)
- the alias restart all php version (eg `for php in $(chkconfig --list | grep php | cut -f1);do sudo service $php restart;done`)

File: salt/lamp/bashrc/files/00-devops.lamp.sh [hunk 7]
```diff
@@ -1,6 +1,6 @@
...
 
-alias cloud9RestartLamp='sudo service httpd restart;sudo service mysql restart;sudo service php-fpm restart'
\ No newline at end of file
+alias cloud9RestartLamp="sudo service httpd restart;sudo service mysql restart;sudo service php-fpm restart"
```
Comment (gervasek) [inline salt/lamp/bashrc/files/00-devops.lamp.sh]: cloud9RestartLamp contains the restart of php: `php-fpm: unrecognized service`

File: salt/lamp/mysql/init.sls [hunk 8]
```diff
@@ -55,8 +58,31 @@ percona-toolkit:
...
+  file.symlink:
+    - target: /etc/my.cnf.d
+    - group: ec2-user
+    - user: ec2-user
+    - makedirs : true
+    
+#Symlink log folder
+/home/ec2-user/environment/log/my.cnf.d:
+  file.symlink:
+    - target: /etc/my.cnf.d
```
Comment (gervasek) [inline salt/lamp/mysql/init.sls]: Should be mysql log `/var/log/mysql`

File: salt/lamp/mysql/init.sls [hunk 9]
```diff
@@ -55,8 +58,31 @@ percona-toolkit:
...
+#Symlink conf folder
+/home/ec2-user/environment/conf.d/my.cnf.d:
+  file.symlink:
+    - target: /etc/my.cnf.d
+    - group: ec2-user
+    - user: ec2-user
+    - makedirs : true
+    
+#Symlink log folder
+/home/ec2-user/environment/log/my.cnf.d:
```
Comment (gervasek) [inline salt/lamp/mysql/init.sls]: replace symlink name `my.cnf.d` by `mysql`

File: salt/lamp/php-fpm/files/www.conf [hunk 10]
```diff
@@ -1,3 +1,10 @@
+;#############################
+;#  Do not edit this file.   #
+;#  File managed by salt.    #
+;#############################
+[global]
+error_log = /var/log/php-fpm/error-{{ php_version }}.log
```
Comment (gervasek) [inline salt/lamp/php-fpm/files/www.conf]: When I restart php-fpm service, log are still in:
- /var/opt/remi/php71/log/php-fpm/error.log 
- /opt/remi/php56/root/var/log/php-fpm/error.log

File: salt/lamp/bashrc/files/00-devops.lamp.sh (hunk 6 above)
Comment (didaurel) [reply salt/lamp/bashrc/files/00-devops.lamp.sh]: - cloud9RestartPhp : restart all php-fpm version
- cloud9RestartPhpXX : restart phpXX-php-fpm
- cloud9RestartLamp : restart all php-fpm version

File: salt/lamp/bashrc/files/00-devops.lamp.sh (hunk 7 above)
Comment (didaurel) [reply salt/lamp/bashrc/files/00-devops.lamp.sh]: - cloud9RestartPhp : restart all php-fpm version
- cloud9RestartPhpXX : restart phpXX-php-fpm
- cloud9RestartLamp : restart all php-fpm version

File: salt/lamp/mysql/init.sls (hunk 8 above)
Comment (didaurel) [reply salt/lamp/mysql/init.sls]: done

File: salt/lamp/mysql/init.sls (hunk 9 above)
Comment (didaurel) [reply salt/lamp/mysql/init.sls]: done

File: salt/lamp/php-fpm/files/www.conf (hunk 10 above)
Comment (didaurel) [reply salt/lamp/php-fpm/files/www.conf]: forsaken

File: salt/tools/files/drush/00-devops.drush.sh [hunk 11]
```diff
@@ -20,3 +20,7 @@ fi
 
 # https://github.com/drush-ops/drush/issues/2065#issuecomment-452549400
 unset module
+
+{% for php_version in php_versions %}
+alias drush{{ php_version }}="/usr/bin/php{{ php_version}} /usr/local/bin/drush"
```
[the remark this answers is not in the document]
Comment (gervasek) [reply salt/tools/files/drush/00-devops.drush.sh]: OK
```
$ drush71  php-eval "print_r(phpversion());"
7.1.28
$ drush56  php-eval "print_r(phpversion());"                                                                                                   
5.6.40
$ drush  php-eval "print_r(phpversion());"
5.6.40
```

File: salt/lamp/mysql/files/my.cnf.d/00-devops.cnf [hunk 12]
```diff
@@ -10,6 +10,8 @@ user        = mysql
 datadir     = /var/lib/mysql
 user        = mysql
 
+log-error   = /var/log/mysql/mysql_error.log
```
[the remark this answers is not in the document]
Comment (didaurel) [reply salt/lamp/mysql/files/my.cnf.d/00-devops.cnf]: Done with your pull request

**Pull request merged**